test/cpp/aot_inductor/test.py

```diff
@@ -5,39 +5,51 @@
 torch.manual_seed(1337)
 
 class Net(torch.nn.Module):
-    def __init__(self):
+    def __init__(self, device):
         super().__init__()
-        self.fc = torch.nn.Linear(64, 10)
+        self.w_pre = torch.randn(4, 4, device=device)
+        self.w_add = torch.randn(4, 4, device=device)
 
-    def forward(self, x, y):
-        return self.fc(torch.sin(x) + torch.cos(y))
+    def forward(self, x):
+        w_transpose = torch.transpose(self.w_pre, 0, 1)
+        w_relu = torch.nn.functional.relu(w_transpose)
+        w = w_relu + self.w_add
+        return torch.matmul(x, w)
 
 data = {}
 
 for device in ["cpu", "cuda"]:
-    model = Net().to(device=device)
-    x = torch.randn((32, 64), device=device)
-    y = torch.randn((32, 64), device=device)
-    with torch.no_grad():
-        ref_output = model(x, y)
-
-    torch._dynamo.reset()
-    with torch.no_grad():
-        constraints = [
-            dynamic_dim(x, 0) >= 1,
-            dynamic_dim(x, 0) <= 1024,
-            dynamic_dim(x, 0) == dynamic_dim(y, 0),
-        ]
-        model_so_path = aot_compile(model, (x, y), constraints=constraints)
-
-    params = dict(model.named_parameters())
-    data.update({
-        f"model_so_path_{device}": model_so_path,
-        f"inputs_{device}": [x, y],
-        f"outputs_{device}": [ref_output],
-        f"fc_weight_{device}": params["fc.weight"],
-        f"fc_bias_{device}": params["fc.bias"],
-    })
+    for use_runtime_constant_folding in [True, False]:
+        if device == "cpu" and use_runtime_constant_folding:
+            # We do not test runtime const folding for cpu mode.
```

Comment:
We can. The piece we are missing at the moment is the way we store constants (we now directly point our weight to the constant we embed in the .so, we would want to separate that to use different buffer.) It would be more direct and easier if we later split out the constants from .so.